### PR TITLE
Fix - updated BGP neighbor parser to support VRF instead of setting 'default' vrf in all cases

### PR DESCRIPTION
**Pull request by mickopp**
## Description
Added a fix to to BGP neighbor parser to enable VRF parsing

## Motivation and Context
enables VRF parsing in "show bgp all neighbors" for neighbors with local-as no-prepend remove-as configuration. previous always "default" was set for VRF

## Impact (If any)
parses VRF instead of "default" in case of VRF configuration for neighbors with config "local-as <x> no-prepend remove-as" configuration 

## Screenshots:
```
python3 folder_parsing_job.py -o iosxe -c ShowBgpAllNeighbors
2022-03-07T21:26:17: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:26:17: %AETEST-INFO: |                   Starting testcase SuperFileBasedTesting                    |
2022-03-07T21:26:17: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:26:17: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:26:17: %AETEST-INFO: |                            Starting section setup                            |
2022-03-07T21:26:17: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:26:17: %AETEST-INFO: The result of section setup is => PASSED
2022-03-07T21:26:17: %AETEST-INFO: The result of testcase SuperFileBasedTesting is => PASSED
2022-03-07T21:26:17: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:26:17: %AETEST-INFO: |                           Starting testcase iosxe                            |
2022-03-07T21:26:17: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:26:17: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:26:17: %AETEST-INFO: |                            Starting section setup                            |
2022-03-07T21:26:17: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:26:17: %AETEST-INFO: The result of section setup is => PASSED
2022-03-07T21:26:17: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:26:17: %AETEST-INFO: |                     Starting section ShowBgpAllNeighbors                     |
2022-03-07T21:26:17: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:26:17: %AETEST-INFO: +..............................................................................+
2022-03-07T21:26:17: %AETEST-INFO: :                Starting STEP 1: iosxe -> ShowBgpAllNeighbors                 :
2022-03-07T21:26:17: %AETEST-INFO: +..............................................................................+
2022-03-07T21:26:17: %AETEST-INFO: +..............................................................................+
2022-03-07T21:26:17: %AETEST-INFO: :        Starting STEP 1.1: Test Golden -> iosxe -> ShowBgpAllNeighbors        :
2022-03-07T21:26:17: %AETEST-INFO: +..............................................................................+
2022-03-07T21:26:17: %AETEST-INFO: +..............................................................................+
2022-03-07T21:26:17: %AETEST-INFO: :    Starting STEP 1.1.1: Gold -> iosxe -> ShowBgpAllNeighbors -> golden_ou    :
2022-03-07T21:26:17: %AETEST-INFO: :                                    tput1                                     :
2022-03-07T21:26:17: %AETEST-INFO: +..............................................................................+
2022-03-07T21:26:17: %AETEST-INFO: The result of STEP 1.1.1: Gold -> iosxe -> ShowBgpAllNeighbors -> golden_output1 is => PASSED
2022-03-07T21:26:17: %AETEST-INFO: +..............................................................................+
2022-03-07T21:26:17: %AETEST-INFO: :    Starting STEP 1.1.2: Gold -> iosxe -> ShowBgpAllNeighbors -> golden_ou    :
2022-03-07T21:26:17: %AETEST-INFO: :                                    tput2                                     :
2022-03-07T21:26:17: %AETEST-INFO: +..............................................................................+
2022-03-07T21:26:17: %AETEST-INFO: The result of STEP 1.1.2: Gold -> iosxe -> ShowBgpAllNeighbors -> golden_output2 is => PASSED
2022-03-07T21:26:17: %AETEST-INFO: +..............................................................................+
2022-03-07T21:26:17: %AETEST-INFO: :    Starting STEP 1.1.3: Gold -> iosxe -> ShowBgpAllNeighbors -> golden_ou    :
2022-03-07T21:26:17: %AETEST-INFO: :                                    tput3                                     :
2022-03-07T21:26:17: %AETEST-INFO: +..............................................................................+
2022-03-07T21:26:17: %AETEST-INFO: The result of STEP 1.1.3: Gold -> iosxe -> ShowBgpAllNeighbors -> golden_output3 is => PASSED
2022-03-07T21:26:17: %AETEST-INFO: +..............................................................................+
2022-03-07T21:26:17: %AETEST-INFO: :    Starting STEP 1.1.4: Gold -> iosxe -> ShowBgpAllNeighbors -> golden_ou    :
2022-03-07T21:26:17: %AETEST-INFO: :                                    tput4                                     :
2022-03-07T21:26:17: %AETEST-INFO: +..............................................................................+
2022-03-07T21:26:18: %AETEST-INFO: The result of STEP 1.1.4: Gold -> iosxe -> ShowBgpAllNeighbors -> golden_output4 is => PASSED
2022-03-07T21:26:18: %AETEST-INFO: +..............................................................................+
2022-03-07T21:26:18: %AETEST-INFO: :    Starting STEP 1.1.5: Gold -> iosxe -> ShowBgpAllNeighbors -> golden_ou    :
2022-03-07T21:26:18: %AETEST-INFO: :                                    tput5                                     :
2022-03-07T21:26:18: %AETEST-INFO: +..............................................................................+
2022-03-07T21:26:18: %AETEST-INFO: The result of STEP 1.1.5: Gold -> iosxe -> ShowBgpAllNeighbors -> golden_output5 is => PASSED
2022-03-07T21:26:18: %AETEST-INFO: +..............................................................................+
2022-03-07T21:26:18: %AETEST-INFO: :    Starting STEP 1.1.6: Gold -> iosxe -> ShowBgpAllNeighbors -> golden_ou    :
2022-03-07T21:26:18: %AETEST-INFO: :                                    tput6                                     :
2022-03-07T21:26:18: %AETEST-INFO: +..............................................................................+
2022-03-07T21:26:18: %AETEST-INFO: The result of STEP 1.1.6: Gold -> iosxe -> ShowBgpAllNeighbors -> golden_output6 is => PASSED
2022-03-07T21:26:18: %AETEST-INFO: +..............................................................................+
2022-03-07T21:26:18: %AETEST-INFO: :    Starting STEP 1.1.7: Gold -> iosxe -> ShowBgpAllNeighbors -> golden_ou    :
2022-03-07T21:26:18: %AETEST-INFO: :                                    tput7                                     :
2022-03-07T21:26:18: %AETEST-INFO: +..............................................................................+
2022-03-07T21:26:18: %AETEST-INFO: The result of STEP 1.1.7: Gold -> iosxe -> ShowBgpAllNeighbors -> golden_output7 is => PASSED
2022-03-07T21:26:18: %AETEST-INFO: The result of STEP 1.1: Test Golden -> iosxe -> ShowBgpAllNeighbors is => PASSED
2022-03-07T21:26:18: %AETEST-INFO: +..............................................................................+
2022-03-07T21:26:18: %AETEST-INFO: :        Starting STEP 1.2: Test Empty -> iosxe -> ShowBgpAllNeighbors         :
2022-03-07T21:26:18: %AETEST-INFO: +..............................................................................+
2022-03-07T21:26:18: %AETEST-INFO: +..............................................................................+
2022-03-07T21:26:18: %AETEST-INFO: :    Starting STEP 1.2.1: Empty -> iosxe -> ShowBgpAllNeighbors -> empty_ou    :
2022-03-07T21:26:18: %AETEST-INFO: :                                     tput                                     :
2022-03-07T21:26:18: %AETEST-INFO: +..............................................................................+
2022-03-07T21:26:18: %AETEST-INFO: The result of STEP 1.2.1: Empty -> iosxe -> ShowBgpAllNeighbors -> empty_output is => PASSED
2022-03-07T21:26:18: %AETEST-INFO: The result of STEP 1.2: Test Empty -> iosxe -> ShowBgpAllNeighbors is => PASSED
2022-03-07T21:26:18: %AETEST-INFO: The result of STEP 1: iosxe -> ShowBgpAllNeighbors is => PASSED
2022-03-07T21:26:18: %AETEST-INFO: +..........................................................+
2022-03-07T21:26:18: %AETEST-INFO: :                       STEPS Report                       :
2022-03-07T21:26:18: %AETEST-INFO: +..........................................................+
2022-03-07T21:26:18: %AETEST-INFO: STEP 1 - iosxe -> ShowBgpAllNeighbors             Passed
2022-03-07T21:26:18: %AETEST-INFO: STEP 1.1 - Test Golden -> iosxe -> ShowBgpAllNeighborsPassed
2022-03-07T21:26:18: %AETEST-INFO: STEP 1.1.1 - Gold -> iosxe -> ShowBgpAllNeighbors -> golden_output1Passed
2022-03-07T21:26:18: %AETEST-INFO: STEP 1.1.2 - Gold -> iosxe -> ShowBgpAllNeighbors -> golden_output2Passed
2022-03-07T21:26:18: %AETEST-INFO: STEP 1.1.3 - Gold -> iosxe -> ShowBgpAllNeighbors -> golden_output3Passed
2022-03-07T21:26:18: %AETEST-INFO: STEP 1.1.4 - Gold -> iosxe -> ShowBgpAllNeighbors -> golden_output4Passed
2022-03-07T21:26:18: %AETEST-INFO: STEP 1.1.5 - Gold -> iosxe -> ShowBgpAllNeighbors -> golden_output5Passed
2022-03-07T21:26:18: %AETEST-INFO: STEP 1.1.6 - Gold -> iosxe -> ShowBgpAllNeighbors -> golden_output6Passed
2022-03-07T21:26:18: %AETEST-INFO: STEP 1.1.7 - Gold -> iosxe -> ShowBgpAllNeighbors -> golden_output7Passed
2022-03-07T21:26:18: %AETEST-INFO: STEP 1.2 - Test Empty -> iosxe -> ShowBgpAllNeighborsPassed
2022-03-07T21:26:18: %AETEST-INFO: STEP 1.2.1 - Empty -> iosxe -> ShowBgpAllNeighbors -> empty_outputPassed
2022-03-07T21:26:18: %AETEST-INFO: ............................................................
2022-03-07T21:26:18: %AETEST-INFO: The result of section ShowBgpAllNeighbors is => PASSED
2022-03-07T21:26:18: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:26:18: %AETEST-INFO: |                           Starting section cleanup                           |
2022-03-07T21:26:18: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:26:18: %AETEST-INFO: The result of section cleanup is => PASSED
2022-03-07T21:26:18: %AETEST-INFO: The result of testcase iosxe is => PASSED
2022-03-07T21:26:18: %GENIE-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:26:18: %GENIE-INFO: |                               Unittest results                               |
2022-03-07T21:26:18: %GENIE-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:26:18: %GENIE-INFO:  SECTIONS/TESTCASES                                                      RESULT
2022-03-07T21:26:18: %GENIE-INFO: --------------------------------------------------------------------------------
2022-03-07T21:26:18: %GENIE-INFO: .
2022-03-07T21:26:18: %GENIE-INFO: |-- SuperFileBasedTesting                                                 PASSED
2022-03-07T21:26:18: %GENIE-INFO: |   `-- setup                                                             PASSED
2022-03-07T21:26:18: %GENIE-INFO: `-- iosxe                                                                 PASSED
2022-03-07T21:26:18: %GENIE-INFO:     |-- setup                                                             PASSED
2022-03-07T21:26:18: %GENIE-INFO:     |-- ShowBgpAllNeighbors                                               PASSED
2022-03-07T21:26:18: %GENIE-INFO:     |   |-- Step 1: iosxe -> ShowBgpAllNeighbors                          PASSED
2022-03-07T21:26:18: %GENIE-INFO:     |   |-- Step 1.1: Test Golden -> iosxe -> ShowBgpAllNeighbors         PASSED
2022-03-07T21:26:18: %GENIE-INFO:     |   |-- Step 1.1.1: Gold -> iosxe -> ShowBgpAllNeighbors -> gol...    PASSED
2022-03-07T21:26:18: %GENIE-INFO:     |   |-- Step 1.1.2: Gold -> iosxe -> ShowBgpAllNeighbors -> gol...    PASSED
2022-03-07T21:26:18: %GENIE-INFO:     |   |-- Step 1.1.3: Gold -> iosxe -> ShowBgpAllNeighbors -> gol...    PASSED
2022-03-07T21:26:18: %GENIE-INFO:     |   |-- Step 1.1.4: Gold -> iosxe -> ShowBgpAllNeighbors -> gol...    PASSED
2022-03-07T21:26:18: %GENIE-INFO:     |   |-- Step 1.1.5: Gold -> iosxe -> ShowBgpAllNeighbors -> gol...    PASSED
2022-03-07T21:26:18: %GENIE-INFO:     |   |-- Step 1.1.6: Gold -> iosxe -> ShowBgpAllNeighbors -> gol...    PASSED
2022-03-07T21:26:18: %GENIE-INFO:     |   |-- Step 1.1.7: Gold -> iosxe -> ShowBgpAllNeighbors -> gol...    PASSED
2022-03-07T21:26:18: %GENIE-INFO:     |   |-- Step 1.2: Test Empty -> iosxe -> ShowBgpAllNeighbors          PASSED
2022-03-07T21:26:18: %GENIE-INFO:     |   `-- Step 1.2.1: Empty -> iosxe -> ShowBgpAllNeighbors -> em...    PASSED
2022-03-07T21:26:18: %GENIE-INFO:     `-- cleanup                                                           PASSED
2022-03-07T21:26:18: %GENIE-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:26:18: %GENIE-INFO: |                                   Summary                                    |
2022-03-07T21:26:18: %GENIE-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:26:18: %GENIE-INFO:  Number of ABORTED                                                            0
2022-03-07T21:26:18: %GENIE-INFO:  Number of BLOCKED                                                            0
2022-03-07T21:26:18: %GENIE-INFO:  Number of ERRORED                                                            0
2022-03-07T21:26:18: %GENIE-INFO:  Number of FAILED                                                             0
2022-03-07T21:26:18: %GENIE-INFO:  Number of PASSED                                                             2
2022-03-07T21:26:18: %GENIE-INFO:  Number of PASSX                                                              0
2022-03-07T21:26:18: %GENIE-INFO:  Number of SKIPPED                                                            0
2022-03-07T21:26:18: %GENIE-INFO:  Total Number                                                                 2
2022-03-07T21:26:18: %GENIE-INFO:  Success Rate                                                            100.0%
2022-03-07T21:26:18: %GENIE-INFO: --------------------------------------------------------------------------------
2022-03-07T21:26:18: %GENIE-INFO:  Total Passing Unittests                                                      8
2022-03-07T21:26:18: %GENIE-INFO:  Total Failed Unittests                                                       0
2022-03-07T21:26:18: %GENIE-INFO:  Total Errored Unittests                                                      0
2022-03-07T21:26:18: %GENIE-INFO:  Total Unittests                                                              8
2022-03-07T21:26:18: %GENIE-INFO: --------------------------------------------------------------------------------






python3 folder_parsing_job.py -o iosxe -c ShowIpBgpAllNeighbors
2022-03-07T21:27:11: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:27:11: %AETEST-INFO: |                   Starting testcase SuperFileBasedTesting                    |
2022-03-07T21:27:11: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:27:11: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:27:11: %AETEST-INFO: |                            Starting section setup                            |
2022-03-07T21:27:11: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:27:11: %AETEST-INFO: The result of section setup is => PASSED
2022-03-07T21:27:11: %AETEST-INFO: The result of testcase SuperFileBasedTesting is => PASSED
2022-03-07T21:27:11: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:27:11: %AETEST-INFO: |                           Starting testcase iosxe                            |
2022-03-07T21:27:11: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:27:11: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:27:11: %AETEST-INFO: |                            Starting section setup                            |
2022-03-07T21:27:11: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:27:11: %AETEST-INFO: The result of section setup is => PASSED
2022-03-07T21:27:11: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:27:11: %AETEST-INFO: |                    Starting section ShowIpBgpAllNeighbors                    |
2022-03-07T21:27:11: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:27:11: %AETEST-INFO: +..............................................................................+
2022-03-07T21:27:11: %AETEST-INFO: :               Starting STEP 1: iosxe -> ShowIpBgpAllNeighbors                :
2022-03-07T21:27:11: %AETEST-INFO: +..............................................................................+
2022-03-07T21:27:11: %AETEST-INFO: +..............................................................................+
2022-03-07T21:27:11: %AETEST-INFO: :       Starting STEP 1.1: Test Golden -> iosxe -> ShowIpBgpAllNeighbors       :
2022-03-07T21:27:11: %AETEST-INFO: +..............................................................................+
2022-03-07T21:27:11: %AETEST-INFO: +..............................................................................+
2022-03-07T21:27:11: %AETEST-INFO: :    Starting STEP 1.1.1: Gold -> iosxe -> ShowIpBgpAllNeighbors -> golden_    :
2022-03-07T21:27:11: %AETEST-INFO: :                                   output1                                    :
2022-03-07T21:27:11: %AETEST-INFO: +..............................................................................+
2022-03-07T21:27:11: %AETEST-INFO: The result of STEP 1.1.1: Gold -> iosxe -> ShowIpBgpAllNeighbors -> golden_output1 is => PASSED
2022-03-07T21:27:11: %AETEST-INFO: +..............................................................................+
2022-03-07T21:27:11: %AETEST-INFO: :    Starting STEP 1.1.2: Gold -> iosxe -> ShowIpBgpAllNeighbors -> golden_    :
2022-03-07T21:27:11: %AETEST-INFO: :                                   output2                                    :
2022-03-07T21:27:11: %AETEST-INFO: +..............................................................................+
2022-03-07T21:27:11: %AETEST-INFO: The result of STEP 1.1.2: Gold -> iosxe -> ShowIpBgpAllNeighbors -> golden_output2 is => PASSED
2022-03-07T21:27:11: %AETEST-INFO: The result of STEP 1.1: Test Golden -> iosxe -> ShowIpBgpAllNeighbors is => PASSED
2022-03-07T21:27:11: %AETEST-INFO: +..............................................................................+
2022-03-07T21:27:11: %AETEST-INFO: :       Starting STEP 1.2: Test Empty -> iosxe -> ShowIpBgpAllNeighbors        :
2022-03-07T21:27:11: %AETEST-INFO: +..............................................................................+
2022-03-07T21:27:11: %AETEST-INFO: +..............................................................................+
2022-03-07T21:27:11: %AETEST-INFO: :    Starting STEP 1.2.1: Empty -> iosxe -> ShowIpBgpAllNeighbors -> empty_    :
2022-03-07T21:27:11: %AETEST-INFO: :                                    output                                    :
2022-03-07T21:27:11: %AETEST-INFO: +..............................................................................+
2022-03-07T21:27:11: %AETEST-INFO: The result of STEP 1.2.1: Empty -> iosxe -> ShowIpBgpAllNeighbors -> empty_output is => PASSED
2022-03-07T21:27:11: %AETEST-INFO: The result of STEP 1.2: Test Empty -> iosxe -> ShowIpBgpAllNeighbors is => PASSED
2022-03-07T21:27:11: %AETEST-INFO: The result of STEP 1: iosxe -> ShowIpBgpAllNeighbors is => PASSED
2022-03-07T21:27:11: %AETEST-INFO: +..........................................................+
2022-03-07T21:27:11: %AETEST-INFO: :                       STEPS Report                       :
2022-03-07T21:27:11: %AETEST-INFO: +..........................................................+
2022-03-07T21:27:11: %AETEST-INFO: STEP 1 - iosxe -> ShowIpBgpAllNeighbors           Passed
2022-03-07T21:27:11: %AETEST-INFO: STEP 1.1 - Test Golden -> iosxe -> ShowIpBgpAllNeighborsPassed
2022-03-07T21:27:11: %AETEST-INFO: STEP 1.1.1 - Gold -> iosxe -> ShowIpBgpAllNeighbors -> golden_output1Passed
2022-03-07T21:27:11: %AETEST-INFO: STEP 1.1.2 - Gold -> iosxe -> ShowIpBgpAllNeighbors -> golden_output2Passed
2022-03-07T21:27:11: %AETEST-INFO: STEP 1.2 - Test Empty -> iosxe -> ShowIpBgpAllNeighborsPassed
2022-03-07T21:27:11: %AETEST-INFO: STEP 1.2.1 - Empty -> iosxe -> ShowIpBgpAllNeighbors -> empty_outputPassed
2022-03-07T21:27:11: %AETEST-INFO: ............................................................
2022-03-07T21:27:11: %AETEST-INFO: The result of section ShowIpBgpAllNeighbors is => PASSED
2022-03-07T21:27:11: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:27:11: %AETEST-INFO: |                           Starting section cleanup                           |
2022-03-07T21:27:11: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:27:11: %AETEST-INFO: The result of section cleanup is => PASSED
2022-03-07T21:27:11: %AETEST-INFO: The result of testcase iosxe is => PASSED
2022-03-07T21:27:11: %GENIE-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:27:11: %GENIE-INFO: |                               Unittest results                               |
2022-03-07T21:27:11: %GENIE-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:27:11: %GENIE-INFO:  SECTIONS/TESTCASES                                                      RESULT
2022-03-07T21:27:11: %GENIE-INFO: --------------------------------------------------------------------------------
2022-03-07T21:27:11: %GENIE-INFO: .
2022-03-07T21:27:11: %GENIE-INFO: |-- SuperFileBasedTesting                                                 PASSED
2022-03-07T21:27:11: %GENIE-INFO: |   `-- setup                                                             PASSED
2022-03-07T21:27:11: %GENIE-INFO: `-- iosxe                                                                 PASSED
2022-03-07T21:27:11: %GENIE-INFO:     |-- setup                                                             PASSED
2022-03-07T21:27:11: %GENIE-INFO:     |-- ShowIpBgpAllNeighbors                                             PASSED
2022-03-07T21:27:11: %GENIE-INFO:     |   |-- Step 1: iosxe -> ShowIpBgpAllNeighbors                        PASSED
2022-03-07T21:27:11: %GENIE-INFO:     |   |-- Step 1.1: Test Golden -> iosxe -> ShowIpBgpAllNeighbors       PASSED
2022-03-07T21:27:11: %GENIE-INFO:     |   |-- Step 1.1.1: Gold -> iosxe -> ShowIpBgpAllNeighbors -> g...    PASSED
2022-03-07T21:27:11: %GENIE-INFO:     |   |-- Step 1.1.2: Gold -> iosxe -> ShowIpBgpAllNeighbors -> g...    PASSED
2022-03-07T21:27:11: %GENIE-INFO:     |   |-- Step 1.2: Test Empty -> iosxe -> ShowIpBgpAllNeighbors        PASSED
2022-03-07T21:27:11: %GENIE-INFO:     |   `-- Step 1.2.1: Empty -> iosxe -> ShowIpBgpAllNeighbors -> ...    PASSED
2022-03-07T21:27:11: %GENIE-INFO:     `-- cleanup                                                           PASSED
2022-03-07T21:27:11: %GENIE-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:27:11: %GENIE-INFO: |                                   Summary                                    |
2022-03-07T21:27:11: %GENIE-INFO: +------------------------------------------------------------------------------+
2022-03-07T21:27:11: %GENIE-INFO:  Number of ABORTED                                                            0
2022-03-07T21:27:11: %GENIE-INFO:  Number of BLOCKED                                                            0
2022-03-07T21:27:11: %GENIE-INFO:  Number of ERRORED                                                            0
2022-03-07T21:27:11: %GENIE-INFO:  Number of FAILED                                                             0
2022-03-07T21:27:11: %GENIE-INFO:  Number of PASSED                                                             2
2022-03-07T21:27:11: %GENIE-INFO:  Number of PASSX                                                              0
2022-03-07T21:27:11: %GENIE-INFO:  Number of SKIPPED                                                            0
2022-03-07T21:27:11: %GENIE-INFO:  Total Number                                                                 2
2022-03-07T21:27:11: %GENIE-INFO:  Success Rate                                                            100.0%
2022-03-07T21:27:11: %GENIE-INFO: --------------------------------------------------------------------------------
2022-03-07T21:27:11: %GENIE-INFO:  Total Passing Unittests                                                      3
2022-03-07T21:27:11: %GENIE-INFO:  Total Failed Unittests                                                       0
2022-03-07T21:27:11: %GENIE-INFO:  Total Errored Unittests                                                      0
2022-03-07T21:27:11: %GENIE-INFO:  Total Unittests                                                              3
2022-03-07T21:27:11: %GENIE-INFO: --------------------------------------------------------------------------------

```

## Checklist:

- [x] I have updated the changelog.
- [ ] I have updated the documentation (If applicable).
- [x] I have added tests to cover my changes (If applicable).
- [x] All new and existing tests passed.
- [x] All new code passed compilation.

### CODE REVIEW

File: changelog/undistributed/changelog_show_bgp_20220307212947.rst [hunk 1]
```diff
@@ -0,0 +1,7 @@
+--------------------------------------------------------------------------------
+                            Fix
+--------------------------------------------------------------------------------
+* IOSXE
+    * Modified ShowBgpNeighbor:
+        * Updated parsing to support VRF in bgp neighbors cli command instead of always setting 'default' VRF (parser p2_3)
+        * Added another files for unit testing (show ip bgp neighbors all, show bgp neighbors all)
```
Comment (LukasMcClelland) [inline changelog/undistributed/changelog_show_bgp_20220307212947.rst:7]:
```suggestion
```

File: src/genie/libs/parser/iosxe/tests/ShowBgpAllNeighbors/cli/equal/golden_output7_output.txt [hunk 2]
```diff
@@ -0,0 +1,259 @@
+#show bgp all neighbors
```
Comment (LukasMcClelland) [inline src/genie/libs/parser/iosxe/tests/ShowBgpAllNeighbors/cli/equal/golden_output7_output.txt:1]: Thank you for adding tests

Would it be possible to reduce the size of the test files but still test you changes?

**Review by LukasMcClelland — CHANGES_REQUESTED** [review]
Thank you for contributing!

File: src/genie/libs/parser/iosxe/tests/ShowBgpAllNeighbors/cli/equal/golden_output7_output.txt (hunk 2 above)
Comment (mickopp) [reply src/genie/libs/parser/iosxe/tests/ShowBgpAllNeighbors/cli/equal/golden_output7_output.txt:1]: Hi Lukas, thanks for your review.
reducing the testfile size would mean reducing the test to one specific bgp neighbor config for the specific bugfix. 
Currently the test checks three different variants of bgp neighbor configs with the testfiles.
Would this make sense ?

File: changelog/undistributed/changelog_show_bgp_20220307212947.rst (hunk 1 above)
Comment (mickopp) [reply changelog/undistributed/changelog_show_bgp_20220307212947.rst]: Hi Lukas,
I added commit 22f421fe26fb7c3e226aea5c40b87c34f552d222 to remove the line from changelog

File: src/genie/libs/parser/iosxe/tests/ShowBgpAllNeighbors/cli/equal/golden_output7_output.txt (hunk 2 above)
Comment (mickopp) [reply src/genie/libs/parser/iosxe/tests/ShowBgpAllNeighbors/cli/equal/golden_output7_output.txt:1]: sorry for the late reply - didn`t notice that i have to click "resolve conversation" - I commited a shorter test for the fix

Comment (LukasMcClelland) [reply src/genie/libs/parser/iosxe/tests/ShowBgpAllNeighbors/cli/equal/golden_output7_output.txt:1]: Not a problem at all, I am replying late as well! Thank you for reducing the test file size

**Review by LukasMcClelland — APPROVED** [review]
Looks great, thank you for contributing!

**Comment by Taarini** [comment]
You can close this PR and open a new PR with the files you want to add.

                               (or)

You can follow the steps below,

- git log --diff-filter=D --summary
- get the commit id of the files you had deleted.
- git checkout <commitid>~1 path/to/file.ext

**Comment by mickopp** [comment]
Thanks @Taarini - i will close this PR and create a new clean PR for the Fix

**Pull request closed without merging**